frontend/document/credocument.lua

```diff
@@ -283,10 +287,132 @@ function CreDocument:updateColorRendering()
     end
 end
 
+function CreDocument:setHideNonlinearFlows(hide_nonlinear_flows)
+    if hide_nonlinear_flows ~= self.hide_nonlinear_flows then
+        self.hide_nonlinear_flows = hide_nonlinear_flows
+        self._document:setIntProperty("crengine.doc.nonlinear.pagebreak.force", self.hide_nonlinear_flows and 1 or 0)
+        self:cacheFlows()
```

Comment:
This one is not wrapped by `if self.hide_nonlinear_flows then` (thanks for wrapping all the others :) 
Is that wrapping safe enough ? No self.flows to clear to be safer ?

Reply:
This `cacheFlows` will effectively clear `self.flows`, I'll put another test down there.